File: dynamic_programming/uniquePath.py
class Solution:
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        path = [1]*n
        for i in range(m):
            for j in range(n):
                if i>0 and j>0:
                    path[j] += path[j-1]
                    
        return path[-1]

File: dynamic_programming/uniquePath.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<int> path(n, 1);
        for (int i=0; i<m; i++){
            for (int j=0; j<n; j++){
                if (i>0 && j>0)
                    path[j] += path[j-1];
            }
        }
        return path[n-1];
    }
};
